sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -421,6 +422,119 @@ ng_ipv6_addr_t *ng_ipv6_netif_find_best_src_addr(kernel_pid_t pid, const ng_ipv6
     return _match_prefix(pid, dest, true);
 }
 
+/* TODO: put this somewhere more central and L2 protocol dependent */
+#define IID_LEN (8)
+
+static bool _hwaddr_to_iid(uint8_t *iid, const uint8_t *hwaddr, size_t hwaddr_len)
```

Comment:
RPL has an option to spread a prefix via control packets (very similar to the PIO of ND). Nodes receiving this prefix should use it to configure an ipv6 address on their interface.

I thought I can use the interface id to create a unique ipv6 address out of that prefix. Do you have any other suggestions how I can achieve this?